### PR TITLE
fix: Multi-trial visualizations switch from metricType to group string [DET-9896]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Multi-trial experiments with custom metrics groups are getting an API endpoint error from `/api/v1/experiments/{id}/metrics-stream/trials-sample` when the searcher metric, or any currently selected metric, is not a training or validation metric.

`trialsSample` and `trialsSnapshot` have already marked `metricType` deprecated and added a `group` string, so we begin using the group string, and change backend response to use group when `metricType` is not specified.

## Test Plan

Download and run experiment from https://gcloud.determined.ai/det/experiments/2713/visualization/learning-curve

All multi-trial visualizations (Learning Curve, HP Parallel Coordinates, HP Scatter Plots, HP Heat Map) load successfully

On a generic / training metric multi-trial experiment, API requests specify the group in the URL, for example `/trials-snapshot?metricName=loss&group=training&batchesProcessed=7&batchesMargin=10`

The API is compatible with old requests, for example if you replace `group=training` in the previous URL for example `/trials-snapshot?metricName=loss&metricType=METRIC_TYPE_TRAINING&batchesProcessed=7&batchesMargin=10`

## Checklist

- [ ] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by garrett361 — APPROVED** [review]
I was able to verify that the experiment in the test plan gets displayed as expected in a local `devcluster`. I'm not very familiar with this part of the code base, though, so I'd recommend waiting on another approval from a reviewer who is before merging. 

Great work!

**Review by gt2345 — APPROVED** [review]
Frontend part looks good to me, I think only passing `group` instead of `validation, training, group` is a good approach going forward.

File: master/internal/api_experiment.go
Comment (hamidzr) [inline master/internal/api_experiment.go:2051]: use model.MetricGroup?

File: master/internal/api_experiment.go
Comment (hamidzr) [inline master/internal/api_experiment.go:1918]: would model.MetricGroup be a closer fit?

File: master/internal/db/postgres_experiments.go
Comment (hamidzr) [inline master/internal/db/postgres_experiments.go:282]: question: what are the differences between Training, Validation, and this generic version? can this replace the other two?

Also can we use the dedicated type for metric groups throughout?

File: master/pkg/model/experiment.go
Comment (hamidzr) [inline master/pkg/model/experiment.go:534]: why do we change this? we see the legacy metric types as a subset of MetricGroup so I think those should all be able to use the same go type

**Review by hamidzr — COMMENTED** [review]
overall looks good, I added some comments on use of the MetricGroup go type and a question on the differences of various trialssnapshot methods.

File: master/internal/api_experiment.go
Comment (mapmeld) [reply master/internal/api_experiment.go]: ✅ latest commit replaces with `model.MetricGroup`

File: master/internal/db/postgres_experiments.go
Comment (mapmeld) [reply master/internal/db/postgres_experiments.go:282]: I'll use `model.MetricGroup` here too
My understanding is Training reads from `steps.metrics->'avg_metrics'->>'MetricName'`
Validation from `validations.metrics->'validation_metrics'->>'MetricName'`

And the generic has to read from the `metrics` table to filter on `metric_group`; otherwise it is more similar to Training with `metrics.metrics->'avg_metrics'->>'MetricName'`

Comment (hamidzr) [reply master/internal/db/postgres_experiments.go:282]: this helper should help with handling the difference between where you'd find the metrics 

https://github.com/determined-ai/determined/blob/80e0294e1f98d70e1f1a91281f9a3755efba9271/master/pkg/model/experiment.go#L525-L531

we should be able to have all of these read from the `metrics` tables with help from maybe
https://github.com/determined-ai/determined/blob/80e0294e1f98d70e1f1a91281f9a3755efba9271/master/internal/db/postgres_trial_metrics.go#L97-L110  and maybe similar to how [GetMetrics](https://github.com/determined-ai/determined/blob/80e0294e1f98d70e1f1a91281f9a3755efba9271/master/internal/db/postgres_trial_metrics.go#L292) works

otherwise I think everything looks good.

Comment (hamidzr) [reply master/internal/db/postgres_experiments.go:241]: 👍🏼 now that there's only one version of these drop the Generic from the name?

**Review by hamidzr — APPROVED** [review]

File: master/internal/db/postgres_experiments.go
Comment (hamidzr) [reply master/internal/db/postgres_experiments.go:241]: thanks for consolidating these.

File: master/internal/db/postgres_experiments.go
Comment (mapmeld) [reply master/internal/db/postgres_experiments.go:262]: I'm unfamiliar with when/how we use `partition_type`. I've added it to the query in this latest commit

**Pull request merged**
